torch/csrc/autograd/saved_variable.cpp

```diff
@@ -43,6 +43,18 @@ SavedVariable::SavedVariable(const Variable& variable, bool is_output, bool is_i
     saved_version_ = version_counter.current_version();
     is_leaf_ = variable.is_leaf();
     is_output_ = is_output;
+    is_inplace_on_view_ = is_inplace_on_view;
+
+    if(is_inplace_on_view) {
+      TORCH_INTERNAL_ASSERT(!is_leaf_ && is_output);
+      weak_grad_fn_ = variable.grad_fn();
+    }
+
+    if (get_default_hooks()) {
+      save_metadata(variable);
+      register_hooks_(get_default_hooks(), variable);
```

Comment:
`register_hooks_` is really confusing to me because it's not obvious what it does. It is shared between this function (the SavedVariable constructor) and `SavedVariable::register_hooks`, but some of the logic only applies for the SavedVariable::register_hooks case. For example, we don't need to do error checking that hooks have already been set and we also don't need to make a check to see if we should save metadata because we're already saving metadata here in the constructor.

Maybe we should rename it to something more descriptive and have it do just the following?
```
set_hooks_and_pack_data(hooks, data) {
    hooks_ = std::move(hooks);
    at::NoGradGuard guard;
    hooks_->call_pack_hook(saved_original_ ? data.tensor_data() : data);
}
```

Reply:
Makes sense to me, I will adapt the PR.